### PR TITLE
[build] move cmake options to file options.cmake

### PR DESCRIPTION
**Pull request by zhanglongxia**

### CODE REVIEW

**Review by bukepo — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -180,6 +116,7 @@ if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
     include(CPack)
 endif()
 
+option(OTBR_DOC "Build documentation" OFF)
```
Comment (wgtdkp) [inline CMakeLists.txt:119]: Move this to `options.cmake`?

**Review by wgtdkp — APPROVED** [review]
Looks great! :+1:

**Review by jinran-google — APPROVED** [review]
LGTM :+1:

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**